test/functional/p2p_seednode.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+Test seednode interaction with the AddrMan
+"""
+import random
+import time
+
+from test_framework.test_framework import BitcoinTestFramework
+
+ADD_NEXT_SEEDNODE = 10
+
+
+class P2PSeedNodes(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.disable_autoconnect = False
+
+    def test_no_seednode(self):
+        # Check that if no seednode is provided, the node proceeds as usual (without waiting)
+        with self.nodes[0].assert_debug_log(expected_msgs=[], unexpected_msgs=["Empty addrman, adding seednode", f"Couldn't connect to peers from addrman after {ADD_NEXT_SEEDNODE} seconds. Adding seednode"], timeout=ADD_NEXT_SEEDNODE):
+            self.restart_node(0)
+
+    def test_seednode_empty_addrman(self):
+        seed_node = "0.0.0.1"
+        # Check that the seednode is added to m_addr_fetches on bootstrap on an empty addrman
+        with self.nodes[0].assert_debug_log(expected_msgs=[f"Empty addrman, adding seednode ({seed_node}) to addrfetch"], timeout=ADD_NEXT_SEEDNODE):
+            self.restart_node(0, extra_args=[f'-seednode={seed_node}'])
+
+    def test_seednode_addrman_unreachable_peers(self):
+        seed_node = "0.0.0.2"
+        node = self.nodes[0]
+        # Fill the addrman with unreachable nodes
+        for i in range(10):
+            ip = f"{random.randrange(128,169)}.{random.randrange(1,255)}.{random.randrange(1,255)}.{random.randrange(1,255)}"
+            port = 8333 + i
+            node.addpeeraddress(ip, port)
```

Comment:
I got this from here: https://github.com/bitcoin/bitcoin/blob/ebd82fa9fae13d5a7a395800295dd7fd34185b58/test/functional/p2p_addr_relay.py#L120

I think the rationale was that I wanted routable yet unreachable peers, but I think you may be right that there's nothing really preventing an IP on that range from running a node at the time the test is run (yet it may be pretty unlikely, given we also use custom ports)

Reply:
Did a bit of digging.  Looks like the approach may have been inherited from https://github.com/bitcoin/bitcoin/pull/22387/commits/b4ece8a1cda69cc268d39d21bba59c51fa2fb9ed (PR #22387).  The risk of "collision" with a real node is extremely low (for now), so it seems like it could be ok to leave as-is (doesn't feel ideal, but unless a suitable alternate address range is identified, it's what we have).

Tried 0.0.1.{random 1-255}, but the test failed (i.e. with `Unexpected message "Empty addrman, adding seednode" partially matches log`).

Nothing pops out immediately in https://en.wikipedia.org/wiki/Reserved_IP_addresses) as both publicly routable and something we don't disqualify already (e.g. `CNetAddr::IsRoutable()`).

Reply:
Is this going to cause the test to try to open connections to somebody on the internet? I think it is highly undesirable if my ISP sees that I try to open connections to these addresses/ports and is able to deduce that I am running bitcoin tests.

Further connections to these addresses may be slow to fail, thus slowing down this test.

You can use something like `-proxy=127.0.0.1:1` to have all connections to any addresses fail quickly.